src/ripple/protocol/tests/STObject.test.cpp

```diff
@@ -192,8 +183,307 @@ class SerializedObject_test : public beast::unit_test::suite
             unexpected (object3.getFieldVL (sfTestVL) != j, "STObject error");
         }
     }
+
+    // Exercise field accessors
+    void
+    testFields()
+    {
+        testcase ("fields");
+
+        auto const& sf1 = sfSequence;
+        auto const& sf2 = sfExpiration;
+        auto const& sf3 = sfQualityIn;
+        auto const& sf4 = sfSignature;
+        auto const& sf5 = sfPublicKey;
+
+        // read free object
+
+        {
+            auto const st = [&]()
+            {
+                STObject st(sfGeneric);
+                st.setFieldU32(sf1, 1);
```

Comment:
Yes that would work, but the point of this initialization is to use routines known to work (and not the code being tested).